enterprise/backend/src/metabase_enterprise/impersonation/middleware.clj

```diff
@@ -13,3 +13,13 @@
                  (lib.metadata/database (qp.store/metadata-provider)))]
     (assoc query :impersonation/role role)
     query))
+
+(defenterprise apply-impersonation-postprocessing
+  "Post-processing middleware. Binds the dynamic var"
+  :feature :advanced-permissions
+  [qp]
+  (fn [query rff]
+    (if-let [role (:impersonation/role query)]
+      (binding [impersonation.driver/*impersonation-role* role]
```

Comment:
Still wonder where it makes sense to use the qp store vs a dynamic var, but can always change this decision later.